promql/value.go

```diff
@@ -168,6 +168,23 @@ func (p HPoint) MarshalJSON() ([]byte, error) {
 	return json.Marshal([...]interface{}{float64(p.T) / 1000, h})
 }
 
+// histogramSize returns the number of samples representing the histogram.
+// The total size is calculated considering the histogram timestamp (p.T - 8 bytes),
+// and then a number of bytes in the histogram.
+// This sum is divided by 16, as samples are 16 bytes.
+func (p HPoint) histogramSize() int {
+	return (p.H.Size() + 8) / 16
```

Comment:
I think we need + 16 here because we also have to take into account the pointer to the `FloatHistogram` (8 bytes).

Reply:
but `Size()` already has the size of pointer, right?

Reply:
if `H` is a pointer, it would return the size of a pointer (8 bytes on a 64bit architecture).
if `H` is not a pointer, it returns the size of the actual thing, but not including the size a pointer would have if it pointed to the thing.

For a float sample, we store the float64 directly, but for histogram samples, we store a pointer to the histogram _plus_ the histogram itself.

Reply:
Sorry, now I get it, you mean you designed the `Size` method such that it already includes the pointer size.

Then it makes sense.